### PR TITLE
Add disable_cleaner option

### PR DESCRIPTION
**Pull request by robertbaldyga**

### CODE REVIEW

**Comment by bkarpins** [comment]
retest this please

**Comment by bkarpins** [comment]
retest this please

**Comment by bkarpins** [comment]
retest this please

File: src/metadata/metadata.c [hunk 1]
```diff
@@ -668,6 +669,8 @@ int ocf_metadata_init_variable_size(struct ocf_cache *cache,
 		/* Re-initialize metadata with different cache line size */
 		ocf_metadata_config_init(cache, line_size);
 
+//	cache->metadata.cleaner_disabled = cleaner_disabled;
+
```
Comment (arutk) [inline src/metadata/metadata.c:673]: commented code

File: doc/requirements/disable_cleaner [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+the cache stop.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Metadata "cleaning" section allocation
+id: cleaning_section_alocation
+---
+
+When diable_cleaner option is selected the "cleaning" section in OCF metadata
+shall not be allocated neiter in DRAM nor on the cache drive.
```
Comment (arutk) [inline doc/requirements/disable_cleaner:41]: *neither

File: src/mngt/ocf_mngt_flush.c [hunk 3]
```diff
@@ -1035,6 +1035,9 @@ void ocf_mngt_cache_cleaning_set_policy(ocf_cache_t cache,
 	if (ocf_cache_is_standby(cache))
 		OCF_CMPL_RET(priv, -OCF_ERR_CACHE_STANDBY);
 
+	if (cache->conf_meta->cleaner_disabled)
+		OCF_CMPL_RET(priv, -OCF_ERR_CLEANER_DISABLED);
```
Comment (arutk) [inline src/mngt/ocf_mngt_flush.c:1039]: requirements suggest, that it should be possible to set policy to NOP when cleaner is disabled. maybe move this check past line 1048.

**Review by arutk — CHANGES_REQUESTED** [review]

File: src/metadata/metadata.c (hunk 1 above)
Comment (robertbaldyga) [reply src/metadata/metadata.c]: Removed.

File: doc/requirements/disable_cleaner (hunk 2 above)
Comment (robertbaldyga) [reply doc/requirements/disable_cleaner]: Fixed.

File: src/mngt/ocf_mngt_flush.c (hunk 3 above)
Comment (robertbaldyga) [reply src/mngt/ocf_mngt_flush.c]: Sounds good. Done.

**Review by arutk — DISMISSED** [review]

**Comment by robertbaldyga** [comment]
retest this please

**Comment by bkarpins** [comment]
retest this please

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    pass_criteria:
+      - Cache attaches properly.
+      - The "cleaning" metadata section is not allocated.
+      - Cache stops properly.
+    steps:
+      - Start the cache with default config.
+      - Prepare default attach config and set cleaner_disabled field to true.
+      - Attach cache device using prepared config.
+      - Verify that cache was attached properly.
+      - Verify that "cleaning" metadata section was not allocated.
```
Comment (arutk) [inline tests/functional/tests/management/test_disable_cleaner.py:25]: how? maybe we should check ~20% smaller metadata memory footprint?

File: tests/functional/tests/management/test_disable_cleaner.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    """
+    pass
+
+
+@pytest.mark.skip(reason="not implemented")
+def test_cleaner_disabled_nop(pyocf_ctx):
+    """
+    title: NOP enfocement in cleaner_disabled mode..
+    description: |
+      Check that after attaching cache wiht cleaner_diabled optios set, the
```
Comment (arutk) [inline tests/functional/tests/management/test_disable_cleaner.py:68]: *with 
*option

**Review by arutk — CHANGES_REQUESTED** [review]

File: tests/functional/tests/management/test_disable_cleaner.py (hunk 4 above)
Comment (robertbaldyga) [reply tests/functional/tests/management/test_disable_cleaner.py:25]: Currently the most reliable method is parsing the log. I wouldn't count on those 20%, as metadata contents may change in the future and this number may become different.

File: tests/functional/tests/management/test_disable_cleaner.py (hunk 5 above)
Comment (robertbaldyga) [reply tests/functional/tests/management/test_disable_cleaner.py]: Fixed.

**Review by arutk — APPROVED** [review]

**Comment by bkarpins** [comment]
retest this please

**Review by mmichal10 — APPROVED** [review]

**Pull request merged**